src/bal-converter/index.ts

```diff
@@ -0,0 +1,178 @@
+  return {
+    banID,
+    banCommonTopoID,
+  };
+};
+
+const balAddrToBanAddr = (
+  balAdresse: BalAdresse,
+  oldBanAddress?: BanAddress
+): BanAddress => {
+  const { banID, banCommonTopoID } = getIds(balAdresse);
+  return {
+    ...(oldBanAddress || {}),
+    id: banID,
+    codeCommune: balAdresse.commune_insee,
+    idVoie: banCommonTopoID,
+    numero: balAdresse.numero,
+    positions: [
+      // Old positions
+      ...(oldBanAddress?.positions || []),
+      {
+        type: balAdresse.position,
+        geometry: {
+          type: "Point",
+          coordinates: [balAdresse.long, balAdresse.lat],
+        },
+      },
+    ],
+    dateMAJ: balAdresse.date_der_maj,
+  };
+};
+
+const balTopoToBanTopo = (
+  balAdresse: BalAdresse,
+  oldBanCommonToponym?: BanCommonToponym
+): BanCommonToponym => {
+  const { banCommonTopoID } = getIds(balAdresse);
+  const isoCodeFromBalNomVoie = (key: LangISO639v3) => key.trim().split("_")[2];
+  const labels = {
+    fr: balAdresse.voie_nom,
+    ...Object.fromEntries(
+      (
+        Object.keys(balAdresse).filter((key) =>
+          key.startsWith("voie_nom_")
+        ) as VoieNomForIsoCode[]
+      ).map((key) => [isoCodeFromBalNomVoie(key), balAdresse[key]])
+    ),
+  };
+
+  return {
+    ...(oldBanCommonToponym || {}),
+    id: banCommonTopoID,
+    codeCommune: balAdresse.commune_insee,
+    label: Object.entries(labels).map(([isoCode, value]) => ({
+      isoCode,
+      value,
+    })), // TODO: rename key to 'labels'
+    type: { value: "voie" }, // TODO: How to get the type from the BAL?
+    geometry: {
+      type: "Point",
+      coordinates: [balAdresse.long, balAdresse.lat],
+    },
+    dateMAJ: balAdresse.date_der_maj,
+  };
+};
+
+const balToBan = (bal: Bal): Ban => {
+  const ban = bal.reduce(
+    (acc: Ban, balAdresse: BalAdresse) => {
+      const { banID, banCommonTopoID } = getIds(balAdresse);
+      const banIdContent = balAddrToBanAddr(balAdresse, acc.addresses?.[banID]);
+      const banCommonTopoIdContent = balTopoToBanTopo(
+        balAdresse,
+        acc.commonToponyms?.[banCommonTopoID]
+      );
+      const districtID = banIdContent.codeCommune;
+      return {
+        ...acc,
+        districtID,
+        addresses: {
+          ...acc.addresses,
+          [banID]: banIdContent,
+        },
+        commonToponyms: {
+          ...acc.commonToponyms,
+          [banCommonTopoID]: banCommonTopoIdContent,
+        },
+      };
+    },
+    {
+      districtID: "",
+      // districts: {},
+      addresses: {},
+      commonToponyms: {},
+    }
+  );
+
+  return ban;
+};
+
+export const sendBalToBan = async (bal: string) => {
+  const balJSON = csvToJson(bal) as Bal;
+  const { districtID, addresses, commonToponyms } = balToBan(balJSON);
+
+  const banAddressIds: BanID[] = Object.keys(addresses || {});
+  const banAddresses: BanAddresses = Object.values(addresses || {});
+  const addressIdsReport = await getAddressIdsReport(districtID, banAddressIds);
```

Comment:
```suggestion
  const addressIdsReportResponse = await getAddressIdsReport(districtID, banAddressIds);
  const addressIdsReport = addressIdsReportResponse.response
```

La réponse de l'API POST /delta-report est de type : 

```
{
    "date": "2023-05-03T13:40:14.134Z",
    "status": "success",
    "response": {
        "idsToCreate": [
            "93b746df-63a1-4904-88c1-2850bcbd9840"
        ],
        "idsToUpdate": [],
        "idsToDelete": [],
        "idsUnauthorized": [
            "93b746df-63a1-4904-88c1-2850bcbd9840"
        ]
    }
}
```

Il faudrait peut-être intercaler le http handler ici pour gérer les réponses (peut-être pas sur cette PR)